packages/core/src/checkbox/checkbox-group.ts

```diff
@@ -1,42 +1,73 @@
 import type { CSSResultGroup, TemplateResult } from 'lit';
+import type { Validator } from '../utils/index.js';
 import type { Checkbox } from './checkbox.js';
+import { MutationController } from '@lit-labs/observers/mutation_controller.js';
 import { LitElement, html } from 'lit';
-import { queryAssignedElements } from 'lit/decorators.js';
-import { EventsController, RovingTabindexController } from '../utils/controllers/index.js';
-import { HintMixin } from '../utils/form-control/index.js';
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import {
+  EventsController,
+  RovingTabindexController,
+  ValidationController,
+  validationStyles
+} from '../utils/controllers/index.js';
+import { HintMixin } from '../utils/mixins/index.js';
+import { requiredValidator } from '../utils/index.js';
 import styles from './checkbox-group.scss.js';
 
 export class CheckboxGroup extends HintMixin(LitElement) {
   /** @private */
   static formAssociated = true;
 
   /** @private */
-  static override styles: CSSResultGroup = styles;
+  static override styles: CSSResultGroup = [validationStyles, styles];
 
   /** Events controller. */
-  #events = new EventsController(this);
+  #events = new EventsController(this, { click: this.#onClick });
 
+  /** Observe changes to the checkboxes. */
+  #mutation = new MutationController(this, {
+    callback: () => {
+      const value = this.boxes
+        ?.map(box => (box.checked ? box.value : null))
+        .filter(Boolean)
+        .join(', ');
+
+      this.#validation.validate(value);
+    },
+    config: { attributeFilter: ['checked'], attributeOldValue: true, subtree: true }
+  });
+
+  /** Manage the keyboard navigation. */
   #rovingTabindexController = new RovingTabindexController<Checkbox>(this, {
     focusInIndex: (elements: Checkbox[]) => elements.findIndex(el => !el.disabled),
```

Comment:
What do you think about focusing all the checkbox element with text just like in this example (not only checkbox element) https://www.w3.org/WAI/ARIA/apg/example-index/checkbox/checkbox?

Reply:
I think that's more of a UI design question. Since checkbox is still being designed, let's wait for that :)